### PR TITLE
test: add wif for github actions

### PR DESCRIPTION
**Pull request by jithumkunjuraman**
Adding WIF for running E2E tests only from the source repo, no forks allowed for now

### CODE REVIEW

File: .github/workflows/build-deploy.yml [hunk 1]
```diff
@@ -38,9 +43,40 @@ jobs:
...
-      - name: Test
+      - name: Run Unit and Integration Test
         run: make verify test
 
+      - name: Auth to GCP account using WIF for E2E tests
+        if: github.repository == github.event.pull_request.head.repo.full_name
+        uses: "google-github-actions/auth@8254fb75a33b976a221574d287e93919e6a36f70"
+        with:
+          project_id: ${{ env.E2EPROJECTID }}
+          workload_identity_provider: "projects/379706009135/locations/global/workloadIdentityPools/gh-gcp-config-connector-tagging-/providers/gh-gcp-config-connector-tagging-" # "projects/123456789/locations/global/workloadIdentityPools/github/providers/my-repo"
```
Comment (muderjul) [inline .github/workflows/build-deploy.yml:54]: Should we also add the project-id as an env var? maybe also the pool name? Not sure how modular this needs to be

**Review by muderjul — APPROVED** [review]

File: .github/workflows/build-deploy.yml [hunk 2]
```diff
@@ -26,8 +27,12 @@ jobs:
   #         version: v1.60.3
   #         args: --timeout 600s --verbose ./...
 
-  test:
+  Test-Operator:
```
Comment (mKeRix) [inline .github/workflows/build-deploy.yml:30]: This job now uses a different naming pattern than the rest - might be nice to have everything consistent, but I'm also ok with it if you'd like to keep it that way

**Review by mKeRix — APPROVED** [review]

File: .github/workflows/build-deploy.yml (hunk 1 above)
Comment (jithumkunjuraman) [reply .github/workflows/build-deploy.yml]: Done

File: .github/workflows/build-deploy.yml (hunk 2 above)
Comment (jithumkunjuraman) [reply .github/workflows/build-deploy.yml]: Thx, changed it back to test.

**Review by muderjul — APPROVED** [review]

**Pull request merged**